sdk/cosmosdb/cosmos/test/public/functional/clientSideEncryption.spec.ts

```diff
@@ -0,0 +1,2346 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { randomUUID } from "@azure/core-util";
+import type { Suite } from "mocha";
+import type {
+  Database,
+  Container,
+  ContainerDefinition,
+  OperationInput,
+  PatchOperation,
+} from "../../../src";
+import {
+  CosmosClient,
+  EncryptionKeyWrapMetadata,
+  ClientEncryptionPolicy,
+  KeyEncryptionKeyAlgorithm,
+  ClientEncryptionIncludedPath,
+  EncryptionType,
+  EncryptionAlgorithm,
+  EncryptionKeyResolverName,
+  StatusCodes,
+  BulkOperationType,
+  ChangeFeedStartFrom,
+  EncryptionQueryBuilder,
+  PatchOperationType,
+  PartitionKeyDefinitionVersion,
+  ChangeFeedMode,
+  ChangeFeedPolicy,
+  ChangeFeedRetentionTimeSpan,
+  PartitionKeyKind,
+  PermissionMode,
+} from "../../../src";
+import { masterKey } from "../common/_fakeTestSecrets";
+import { endpoint } from "../common/_testConfig";
+import {
+  compareMetadata,
+  MockKeyVaultEncryptionKeyResolver,
+  testCreateClientEncryptionKey,
+  testCreateItem,
+  TestDoc,
+  verifyExpectedDocResponse,
+  verifyItemByRead,
+  testRewrapClientEncryptionKey,
+  testUpsertItem,
+  validateQueryResults,
+  testPatchItem,
+  testReplaceItem,
+  testDeleteItem,
+  verifyDiagnostics,
+} from "../common/encryptionTestHelpers";
+import { removeAllDatabases } from "../common/TestHelpers";
+import { assert } from "chai";
+import { EncryptionTimeToLive } from "../../../src/encryption/EncryptionTimeToLive";
+
+let encryptionClient: CosmosClient;
+let metadata1: EncryptionKeyWrapMetadata;
+let metadata2: EncryptionKeyWrapMetadata;
+let database: Database;
+let encryptionContainer: Container;
+let encryptionContainerForChangeFeed: Container;
+let testKeyEncryptionKeyResolver: MockKeyVaultEncryptionKeyResolver;
+let containerDefinition: ContainerDefinition;
+let clientEncryptionPolicy: ClientEncryptionPolicy;
+
+const testKeyVault = "TESTKEYSTORE_VAULT" as EncryptionKeyResolverName;
+
+describe("Client Side Encryption", function (this: Suite) {
+  before(async () => {
+    await removeAllDatabases();
+    testKeyEncryptionKeyResolver = new MockKeyVaultEncryptionKeyResolver();
+    metadata1 = new EncryptionKeyWrapMetadata(
+      testKeyVault,
+      "key1",
+      "tempmetadata1",
```

Comment:
Could you clarify how tempmetadata1 relates to the CMK path? It's a bit unclear. Are you referring to the HTTP path of the key in Key Vault?